Place Postprocessor Files Within Respective "postprocessor_output" Subdirectories

danieldouglas92 opened the pull request:
This is something that I don't feel is absolutely necessary, but it is something that I've thought would make sense. I often found that for models that I ran which output topography I would personally make a topography directory and just move the files there, so I figured this change might also be appreciated by others. If this does seem worth while, there are several other postprocessors which could benefit from a similar change.

danieldouglas92 committed:
create topography subdirectory

bangerth reviewed (commented):
This is ok with me, but let's say what others have to say. You'll have to adjust tests as well, and you should check that the test harness doesn't just pick up stored output file in the top level directory of every test, but recurses into sub-directories as well.

gassmoeller reviewed (commented):
Yes, I like it! There are several postprocessors (e.g. also heat flux map) that just flood the output directory with files, and I am much in favor of separating them into different directories. As Wolfgang said you will have to update test results.

danieldouglas92 committed:
update tests

gassmoeller reviewed (approved):
Looks good to me, if the test I suggested above works (=fails) I think this is ready to merge.

danieldouglas92 commented:
@gassmoeller I rebased and fixed the test output. If it makes sense I can go through and modify the rest of the Postprocessors in this PR? Otherwise I'll open a follow-up PR once this one is merged that does this for the other Postprocessors.

gassmoeller reviewed (approved):
Lets do the other postprocessors in a follow up PR. Keeping PRs small and separate makes it easier to review and figure out changes in the future.

bangerth commented:
I'm sorry to take a while to get to this. The check for which files to compare indeed recurses into subdirectories, as I think you've found out above:
https://github.com/geodynamics/aspect/blob/main/tests/CMakeLists.txt#L238-L239
So I think it's all good!
